service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/ObservationStatementMapper.java

```diff
@@ -91,6 +104,37 @@ private String prepareComment(Observation observation) {
                     pertinentInformationObservationValueMapper.mapReferenceRangeToPertinentInformation(referenceRange));
             }
         }
+
+        if (observation.hasInterpretation()) {
+            CodeableConcept interpretation = observation.getInterpretation();
+
+            boolean isInterpretationCode = interpretation.getCoding().stream()
+                .anyMatch(this::isInterpretationCode);
+
+            if (!isInterpretationCode) {
+                Optional<Coding> codeWithUserSelected = interpretation.getCoding().stream()
+                    .filter(code -> !isInterpretationCode(code) && code.hasUserSelected() && code.getUserSelected())
+                    .findFirst();
+
+                Coding coding = codeWithUserSelected.orElse(interpretation.getCodingFirstRep());
```

Comment:
I think at this point we're still not guaranteed to have any codings. In that case, will `getCodingFirstRep()` just return `null` or throw an exception?

Reply:
From what I can gather, it looks as though `getCodingFirstRep()` invokes `addCoding()` when it's empty. 

`addCoding()` looks like it creates a new `ArrayList<Coding>` if it's null, and adds a new `Coding()` object, so I don't believe it will return null, but I'm not 100% certain on that.

Reply:
Yes. And then none of the `if`s below will pass and no further action will be taken. There is still the side-effect of the `observation` being changed, i.e. being given an empty coding, but it looks like they take some effect to ignore `isEmpty` codings. I would feel more comfortable being explicit about it though...

```suggestion
                Coding coding = codeWithUserSelected.orElseGet(() -> interpretation.hasCoding()
                    ? interpretation.getCodingFirstRep()
                    : null);
```

Reply:
I like this solution, though doing this would mean that I'd need to have a `if (coding != null)` condition creating the `pertinentInformation`

Would using `Optional.ofNullable` be an OK solution to this issue so that it can be more explicit? 

```suggestion
                Coding coding = codeWithUserSelected.orElseGet(() ->
                    interpretation.hasCoding() ? interpretation.getCodingFirstRep() : null);

                Optional.ofNullable(coding).ifPresent(codeable -> {
                    if (codeable.hasDisplay()) {
                        commentBuilder.insert(COMMENT_OFFSET, codeable.getDisplay() + StringUtils.SPACE);
                    }
                    if (codeable.hasCode()) {
                        commentBuilder.insert(COMMENT_OFFSET, coding.getCode() + StringUtils.SPACE);
                    }
                    if (codeable.hasCode() || codeable.hasDisplay()) {
                        commentBuilder.insert(COMMENT_OFFSET, "Interpretation Code: ");
                    }

                    if (interpretation.hasText()) {
                        commentBuilder.insert(COMMENT_OFFSET, "Interpretation Text: "
                            + interpretation.getText() + StringUtils.SPACE);
                    }
                });
```

Reply:
That's fine, or make it an optional in the first place

Reply:
Done in 27391b0